core/trino-main/src/main/java/io/trino/sql/planner/planprinter/PlanPrinter.java

```diff
@@ -266,6 +272,11 @@ public static String textDistributedPlan(
                 .flatMap(Collection::stream)
                 .collect(toImmutableMap(Entry::getKey, Entry::getValue));
 
+        Map<PlanNodeId, Metrics> connectorMetrics = queryStats.getOperatorSummaries().stream()
+                .filter(stats -> stats.getOperatorType().contains("Scan")) // sometimes >1 operators having the same plan node ID
```

Comment:
`connectorMetrics/metrics` should be stored within `io.trino.sql.planner.planprinter.PlanNodeStats` and should go normal plan node stats route.

Reply:
Good point, thanks!